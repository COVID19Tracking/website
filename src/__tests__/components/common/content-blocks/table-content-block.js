import React from 'react'
import renderer from 'react-test-renderer'
import TableContentBlock from '~components/common/content-blocks/table-content-block'

describe('Components : Common : Table Content Block', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <TableContentBlock
          table={`| test  | header | this  | is    | header |
          |-------|--------|-------|-------|--------|
          | 1,000 | 2,000  | 3,000 | 4,000 | 5,000  |
          | N/A   | N/A    | This  | is c  | a cell |
          | Woot  | De     | Doot  | De    | Doo    |`}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
