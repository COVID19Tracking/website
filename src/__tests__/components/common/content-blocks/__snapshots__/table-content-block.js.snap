// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common : Table Content Block renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr
      className="header"
    >
      <th
        dangerouslySetInnerHTML={
          Object {
            "__html": "test",
          }
        }
      />
      <th
        dangerouslySetInnerHTML={
          Object {
            "__html": "header",
          }
        }
      />
      <th
        dangerouslySetInnerHTML={
          Object {
            "__html": "this",
          }
        }
      />
      <th
        dangerouslySetInnerHTML={
          Object {
            "__html": "is",
          }
        }
      />
      <th
        dangerouslySetInnerHTML={
          Object {
            "__html": "header",
          }
        }
      />
    </tr>
  </thead>
  <tbody>
    <tr
      className="row"
    >
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "test",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<hr>",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "this",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<hr>",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "is",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<hr>",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<hr>",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<hr>",
            }
          }
        />
      </td>
    </tr>
    <tr
      className="row"
    >
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "test",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "1,000",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "this",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "2,000",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "is",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "3,000",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "4,000",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "5,000",
            }
          }
        />
      </td>
    </tr>
    <tr
      className="row"
    >
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "test",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "N/A",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "this",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "N/A",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "is",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "This",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "is c",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "a cell",
            }
          }
        />
      </td>
    </tr>
    <tr
      className="row"
    >
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "test",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Woot",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "this",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "De",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "is",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Doot",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": "header",
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "De",
            }
          }
        />
      </td>
      <td
        alignLeft={true}
      >
        <span
          className="cellLabel"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Doo",
            }
          }
        />
      </td>
    </tr>
  </tbody>
</table>
`;
